File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int ans = 0;
        int left = 0;
        int right = height.size()-1;
          
          while(left <= right){
         int breath = right - left;
         int ht = min(height[left] , height[right]);
         int wt = breath * ht;
         ans = max(ans , wt);

         if(height[left] < height[right]) left++;
         else right--;
          }
            return ans;
           }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        left = 0
        right = len(height) - 1
        water = 0

        while left <= right:
            length = right - left
            ht = min(height[left] , height[right])
            area = ht * length
            water = max(water , area)

            if(height[left] < height[right]): left += 1
            else: right -= 1


        return water
        